mentii/mentii
# Problem from class

This story set up the flow of getting a problem from a textbook when you click on do problem within an activity. The main flow is: 
 - User pushes a button that sends the class code and activity name to the backend
 - A database lookup to get find what book, chapter, and section the activity is using.
 - A problem is randomly selected from the problems in the appropriate section.
 - The problem is used to generate a tree.
 - That tree is sent back to the front end and user. 

Any errors in this process result in the same error message that you get when a problem is not returned currently. 

**Significant Changes**
1. Activities now should include a bookId, chapterTitle, and sectionTitle
2. The sections within chapters within books are the only place problem templates are stored
3. There is a button in all activities shown labeled "do problem"
4. The problem that is selected is randomly chosen based off of what is in the book section for that activity.

**How to Test**
1. Add a book containing at least one chapter, section, and problem within that section to the system.
2. Create at least one class in the system. 
3. Directly in the database, update the activity in that class to have a bookId, chapterTitle, and sectionTitle field corresponding to the fields in the book you created. 
4. Login and navigate to the class that you created and edited.
5. Join the class and view the class details page. 
6. Click on do problem and see that the problem is displaying and working properly. 
7. In the sample data on staging week 3 shows you what it looks like from the user perspective if there is an error.

**Notes**
This is a little rough to test by hand because you have to edit the database directly. The unit test populates the database with all the data you need and you can use that as an example.
